### PR TITLE
Add pagination for get nfts API + Create get collections API

### PR DESCRIPTION
**Pull request by edisonz0718**
This PR:
1. Add pagination for get nfts API
2. Create get collections API to get nft collection's data
3. Clean up deprecated APIs
4. Update API response format

### CODE REVIEW

File: api/internal/repository/nft_token.go [hunk 1]
```diff
@@ -75,6 +75,14 @@ func (n *nftTokenDbImpl) GetNfts(collectionAddresses []string, walletAddress str
 		query = query.Where("owner_address = ?", walletAddress)
 	}
 
+	if offset != nil {
```
Comment (LeoHChen) [inline api/internal/repository/nft_token.go:78]: any sanity checking on the offset?

File: api/internal/repository/nft_token.go [hunk 2]
```diff
@@ -75,6 +75,14 @@ func (n *nftTokenDbImpl) GetNfts(collectionAddresses []string, walletAddress str
...
+		query = query.Offset(*offset)
+	}
+
+	if limit != nil {
```
Comment (LeoHChen) [inline api/internal/repository/nft_token.go:82]: there has to be an upper bound of the limit

File: api/internal/repository/nft_token.go (hunk 1 above)
Comment (edisonz0718) [reply api/internal/repository/nft_token.go:78]: Added validation to check if offset is > 0 in the handler. For offset that is larger than the total db record, the query will return empty array.

File: api/internal/repository/nft_token.go (hunk 2 above)
Comment (edisonz0718) [reply api/internal/repository/nft_token.go:82]: Yes. Added validation in the handler to make sure the limit is > 0 and <= 500

**Review by LeoHChen — APPROVED** [review]

**Pull request merged**